### ISSUE TITLE
Definition of CNOT 

### ISSUE THREAD

**Issue by tilias4** [qutip/qutip#2494]
### Bug Description

The act of the CNOT gate is not what it is expected. In particular it changes the target qubit when the control qubit is in the ground state instead of excited state given that in qutip basis(2,1) is the ground state. If this is on purpose it needs to be clearly stated in the tutorials or the documentation

### Code to Reproduce the Bug

```shell
from qutip import *
from qutip_qip.circuit import QubitCircuit
from qutip_qip.operations import (Gate, berkeley, cnot, cphase, csign, fredkin,
                                  gate_sequence_product, globalphase, iswap,
                                  molmer_sorensen, phasegate, qrot, rx, ry, rz,
                                  snot, sqrtiswap, sqrtnot, sqrtswap, swap,
                                  swapalpha, toffoli)
test=cnot()*tensor(basis(2,1),basis(2,1))
print(test)
```

### Code Output

```shell
Quantum object: dims=[[2, 2], [1, 1]], shape=(4, 1), type='ket', dtype=Dense
Qobj data =
[[0.]
 [0.]
 [1.]
 [0.]]
```

### Expected Behaviour

Quantum object: dims=[[2, 2], [1, 1]], shape=(4, 1), type='ket', dtype=Dense
Qobj data =
[[0.]
 [0.]
 [0.]
 [1.]]

### Your Environment

```shell
QuTiP Version:      5.0.2
Numpy Version:      1.26.4
Scipy Version:      1.11.4
Cython Version:     None
Matplotlib Version: 3.8.0
Python Version:     3.11.0
Number of CPUs:     8
BLAS Info:          Generic
INTEL MKL Ext:      False
Platform Info:      Darwin (arm64)
Installation path:  /opt/anaconda3/lib/python3.11/site-packages/qutip
```

**Comment by Ericgig** [comment]
`qutip.basis(N, 0)` is the ground state almost everywhere in qutip.
I am pretty sure it's the same for every gates, but @BoxiLi could confirm.

**Comment by tilias4** [comment]
But if I apply sigmaz()*basis(2,0) it will give the same state with eigenvalue 1. However, it is expected that the application of sigmaz() in the ground state to give me the same state with eigenvalue -1. This is happening with sigmaz()*basis(2,1). So 
sigmaz()*basis(2,1)=-basis(2,1) while sigmaz()*basis(2,0)=+basis(2,0). Doesn’t that mean that basis(2,0) is the excited state and basis(2,1) is the ground state? Am I misunderstanding something here?

**Comment by BoxiLi** [comment]
In qutip `basis(2,0)` represents the ground state. If you understand `sigmaz()` as a gate in quantum information processing, it often refers to accumulating a pi phase for the excited state, which is adding `exp(-i*pi)=-1` to `basis(2,1)`. If you want a more quantum optics definition, you can use `qutip.num` instead.

If you compare the vector representation of state array in qutip with other circuit simulators, some of them use a different convention in qubits ordering. E,g. [0,0,1,0] represents the excitation of the first qubit but in some other software the second qubit.

**Comment by tilias4** [comment]
If basis(2,0) represents the ground state then does the operator sigmam() represents the creation operator?

In the documentation of qutip it says it represents the “Annihilation operator for Pauli spins.” Which means, to my understanding, that if it is applied to the ground state it should give zero. This is happening when it is applied to basis(2,1) and NOT to basis(2,0). Analogously, the sigmap() which according to the documentation it represents the creation operator gives zero when applied to basis(2,0) meaning that basis(2,0) should be the excited state.

**Comment by BoxiLi** [comment]
Yes, you are right, there are different conventions used in physics. If the Pauli matrix is used to represent spins, the operators are defined according to the Pauli commutation relation, see https://easyspin.org/easyspin/documentation/spinoperators.html. Here indeed `basis(2,0)` behaves like the spin-up state.

But if you use quantum information or quantum optics convention, typically `basis(2,0)` is the ground state. You can also see this in the difference between the result of `qutip.create` and `qutip.sigmap`.

A note should probably be added to the doc here.

**Comment by ajgpitch** [comment]
I remember also finding this confusing back in the day. Where in the documentation the documentation do we think we should address this?
